test/common-tests/request-param-validation-api-tests/string-param-validation-api-tests.ts

```diff
@@ -0,0 +1,116 @@
+import { getMinAndMaxLengthFromOptions } from '@ukef-test/support/helpers/min-and-max-length-helper';
+import request from 'supertest';
+
+export interface ParamValidationApiTestOptions {
+  paramName: string;
+  length?: number;
+  minLength?: number;
+  maxLength?: number;
+  pattern?: RegExp;
+  enum?: any;
+  generateParamValueOfLength?: (length: number) => string;
+  generateParamValueThatDoesNotMatchRegex?: () => string;
+  generateParamValueThatDoesNotMatchEnum?: () => string;
+  validRequestParam: string;
+  makeRequest: (paramValue: string) => request.Test;
+  givenAnyRequestParamWouldSucceed: () => void;
+}
+
+export const withParamValidationApiTests = ({
+  paramName: paramNameSymbol,
+  length: lengthOption,
+  minLength: minLengthOption,
+  maxLength: maxLengthOption,
+  pattern,
+  enum: theEnum,
+  generateParamValueOfLength,
+  generateParamValueThatDoesNotMatchRegex,
+  generateParamValueThatDoesNotMatchEnum,
+  validRequestParam,
+  makeRequest,
+  givenAnyRequestParamWouldSucceed,
+}: ParamValidationApiTestOptions) => {
+  const paramName = paramNameSymbol as string;
+  const { minLength, maxLength } = getMinAndMaxLengthFromOptions({ parameterName: paramName, minLengthOption, maxLengthOption, lengthOption });
+
+  describe(`${paramName} validation`, () => {
+    beforeEach(() => {
+      givenAnyRequestParamWouldSucceed();
+    });
+
+    it(`returns a 2xx response if ${paramName} is valid`, async () => {
+      const { status } = await makeRequest(validRequestParam);
+
+      expect(status).toBeGreaterThanOrEqual(200);
+      expect(status).toBeLessThan(300);
+    });
```

Comment:
Can we improve on what we did with TFS and make this `returns a ${successStatusCode} response if ${paramName} is valid` please, where the caller sets the successStatus Code, please?